portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,51 +30,76 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  final int SC_NO_CONTENT = 204;
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      datastore.delete(entity.getKey());
+    }
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Greeting");
+    Query query = new Query("Comment");
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<String> greetings = new ArrayList<>();
+    List<String> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
-      String greeting = (String) entity.getProperty("greeting");
-      greetings.add(greeting);
+      String comment = (String) entity.getProperty("comment");
+
+      String language = request.getParameter("languages");
+
+      if (language != null && !language.equals("original")) {
+        Translate translate = TranslateOptions.getDefaultInstance().getService();
+        Translation translation = translate.translate(comment, Translate.TranslateOption.targetLanguage(language));
```

Comment:
It is better to reduce the number of calls to translation API for better latency. There is a [translate function ](https://googleapis.dev/java/google-cloud-translate/latest/com/google/cloud/translate/Translate.html#translate-java.util.List-com.google.cloud.translate.Translate.TranslateOption...-) that also accepts a list of strings.

Reply:
I couldn't find a getTranslatedText() method for lists so I would have to iterate through the list twice manually if I want to add the elements, translate them, then add the translated back to the list of strings. Wouldn't this make the process longer than the latency in the original case?